needle.pm

```diff
@@ -311,9 +311,13 @@ sub wanted_ {
 sub default_needles_dir { "$bmwqemu::vars{PRODUCTDIR}/needles" }
 
 sub init {
-    $needles_dir = ($bmwqemu::vars{NEEDLES_DIR} // default_needles_dir);
-    $needles_dir = File::Spec->catdir($bmwqemu::vars{CASEDIR}, $needles_dir) unless -d $needles_dir;
-    die "needles_dir not found: $needles_dir (check vars.json?)" unless -d $needles_dir;
+    my $default_needles_dir = ($bmwqemu::vars{NEEDLES_DIR} // default_needles_dir);
+    $needles_dir = $default_needles_dir;
+    unless (-d $needles_dir) {
+        die "Can't init needles from $needles_dir" if (File::Spec->file_name_is_absolute($needles_dir));
+        $needles_dir = File::Spec->catdir($bmwqemu::vars{CASEDIR}, $needles_dir);
+        die "Can't init needles from $default_needles_dir (If user doesn't specify NEEDLES_DIR, the needles will be inited from \$PRODUCTDIR/needles firstly) or $needles_dir (\$CASEDIR + $default_needles_dir) (check vars.json?)" unless -d $needles_dir;
```

Comment:
```suggestion
        die "Can't init needles from $default_needles_dir; if one doesn't specify NEEDLES_DIR, the needles will be loaded from \$PRODUCTDIR/needles firstly or $needles_dir (\$CASEDIR + $default_needles_dir), check vars.json" unless -d $needles_dir;
```

But this still exceeds the limit for the reason. Maybe we should just put a reference to some documentation here.

Reply:
Or point to the log output with an explicit error message.

Reply:
You can drop the mention of `vars.json` so long as the relevant variables are mentioned. Pointing to logs or files here is mostly confusing if you don't know what you're looking for e.g. it won't tell you that a magic folder is expected but missing or in the wrong place.

Reply:
I modified the message according @Martchus comments. 
IMHO the whole error message is shown in autoinst-log.txt, users can see the details in it even the reason is not complete. 
I keep that `vars.json` to reminder users could get the CASEDIR or PRODUCTDIR or NEEDLES_DIR from it. So I prefer to keep that message.

Reply:
Yeah, I think that is ok. Could you please add a test covering the line?

Reply:
added